0xProject/0x-api
# chore: adding expectedSlippage to quote reports [GRO-27]

# Description



Data uses Quote Reports to compare settled trades versus the quotes.

Adding the expectedSlippage into the Quote Report from the data model will make it easier to join the data sets together and ensure the changes are making greater improvements. This change adds expectedSlippage as an optional field in the quote report, so not every report will include it. It will only be included in the report if the quote itself provides an expectedSlippage value.

# Testing Instructions



Setup kafka locally and configure 0xAPI to point to mainnet and your local kafka. Run some quote requests against your local endpoint and verify the quote reports sent to kafka. For most test cases, you will not see expectedSlippage to appear in the quote report sent to kafka, however, unless the quote you requested has that information in the quote itself.

More information on kafka can be found at:
https://www.conduktor.io/kafka/starting-kafka

# Checklist





    For more information see `0x API Matcha smoke test runbook` in Quip.
